base_export_email/ir_actions.py

```diff
@@ -0,0 +1,162 @@
+            data.update({'model': model.model})
+        return {'value': data}
+
+    def _search_data(self, cr, uid, action, context=None):
+        obj_pool = self.pool[action.model]
+        domain = action.filter_id and safe_eval(
+            action.filter_id.domain, {'time': time}) or []
+        ctx = action.filter_id and safe_eval(
+            action.filter_id.context) or context
+        return obj_pool.search(cr, uid, domain,
+                               offset=0, limit=False, order=False,
+                               context=ctx)
+
+    def _export_data(self, cr, uid, obj_ids, action, context=None):
+        obj_pool = self.pool[action.model]
+        export_fields = []
+        if action.fields_to_export:
+            export_fields = safe_eval(action.fields_to_export)
+        if action.saved_export_id:
+            # retrieve fields of the selected list
+            export_fields.extend(
+                [efl.name for efl in
+                 action.saved_export_id.export_fields])
+        return export_fields, obj_pool.export_data(
+            cr, uid, obj_ids, export_fields,
+            context=context).get('datas', [])
+
+    def _send_email(self, cr, uid, action, export_fields, export_data,
+                    context=None):
+        """
+        Prepare a message with the exported data to send with the
+        template of the configuration
+        """
+        mail_compose = self.pool['mail.compose.message']
+        values = mail_compose.onchange_template_id(
+            cr, uid, 0, action.email_template_id, 'comment',
+            'ir.actions.server', action.id, context=context)['value']
+        values['partner_ids'] = [
+            (4, partner_id) for partner_id in values.pop('partner_ids',
+                                                         [])
+        ]
+        if action.export_format == 'csv':
+            export = main.CSVExport()
+        else:
+            export = main.ExcelExport()
+        filename = export.filename(action.model)
+        content = export.from_data(export_fields, export_data)
+        if isinstance(content, unicode):
+            content = content.encode('utf-8')
+        data_attach = {
+            'name': filename,
+            'datas': base64.b64encode(str(content)),
+            'datas_fname': filename,
+            'description': filename,
+        }
+
+        values['attachment_ids'] = [(0, 0, data_attach)]
+        compose_id = mail_compose.create(
+            cr, uid, values, context=context)
+        return mail_compose.send_mail(cr, uid, [compose_id], context=context)
+
+    def run(self, cr, uid, ids, context=None):
+        """
+        If the state of an action is export_email,
+        export data related to the configuration and send the result by email
+        """
+        for action in self.browse(cr, uid, ids, context):
+            if action.state == 'export_email':
+                ids.remove(action.id)
```

Comment:
In my opinion it is not usefull to evaluate the expression.
The super, do this evaluation as first step of the for and then dispatch call following the action.state.
I think, it is not usefull to send id of action in state 'export_email' because the case is already processed, so I remove if from the list.